src/Storages/MergeTree/MergeTreeSettings.h

```diff
@@ -133,6 +133,7 @@ struct Settings;
     M(MaxThreads, max_part_removal_threads, 0, "The number of threads for concurrent removal of inactive data parts. One is usually enough, but in 'Google Compute Environment SSD Persistent Disks' file removal (unlink) operation is extraordinarily slow and you probably have to increase this number (recommended is up to 16).", 0) \
     M(UInt64, concurrent_part_removal_threshold, 100, "Activate concurrent part removal (see 'max_part_removal_threads') only if the number of inactive data parts is at least this.", 0) \
     M(String, storage_policy, "default", "Name of storage disk policy", 0) \
+    M(String, disk, "", "Name of storage disk. Can be specified instead of storage policy.", 0) \
```

Comment:
Ok, though this setting for merge tree named "disk" is not only for custom disk, but to allow writing disk name instead of storage policy (e.g. a second feature of this PR). So we need to have both - string value of "disk" setting and function like value of "disk" setting.